python/lsst/meas/extensions/piff/piffPsf.py

```diff
@@ -37,14 +38,24 @@ class PiffPsf(ImagePsf):
         "PiffPsf"
     )
 
-    def __init__(self, width, height, piffResult):
+    def __init__(self, width, height, piffResult, log=None):
         assert width == height
         ImagePsf.__init__(self)
         self.width = width
         self.height = height
         self.dimensions = Extent2I(width, height)
         self._piffResult = piffResult
         self._averagePosition = None
+        self._averageColor = None
+        try:
+            color_type = [s.data.properties['colorType'] for s in self._piffResult.stars
+                          if not s.is_flagged and not s.is_reserve]
+            color_type = list(set(color_type))
+            self._color_type = color_type[0]
```

Comment:
We need to set `_color_type` to be `None` if `color_type` has more than one element and log a warning. So moving the `log` up would be good.

Reply:
What's the motivation to get `color_type` at the time of initialization, but get the value lazily when `getAverageColor` is called? It'd seem fine to do this lazily as well.